src/AzureAppConfigurationImpl.ts

```diff
@@ -196,17 +224,58 @@ export class AzureAppConfigurationImpl implements AzureAppConfiguration {
             keyValues.push([key, value]);
         }
 
-        this.#configMap.clear(); // clear existing key-values in case of configuration setting deletion
+        this.#clearLoadedKeyValues(); // clear existing key-values in case of configuration setting deletion
         for (const [k, v] of keyValues) {
             this.#configMap.set(k, v);
         }
     }
 
+    async #clearLoadedKeyValues() {
+        for(const key of this.#configMap.keys()) {
+            if (key !== FEATURE_MANAGEMENT_KEY_NAME) {
+                this.#configMap.delete(key);
+            }
+        }
+    }
+
+    async #loadFeatureFlags() {
+        const featureFlags: FeatureFlagValue[] = [];
+        const featureFlagSelectors = getValidSelectors(this.#options?.featureFlagOptions?.selectors);
+        for (const selector of featureFlagSelectors) {
+            const listOptions: ListConfigurationSettingsOptions = {
+                keyFilter: `${featureFlagPrefix}${selector.keyFilter}`,
+                labelFilter: selector.labelFilter
+            };
+            const requestTraceOptions = {
+                requestTracingEnabled: this.#requestTracingEnabled,
+                initialLoadCompleted: this.#isInitialLoadCompleted,
+                appConfigOptions: this.#options
+            };
+            const settings = listConfigurationSettingsWithTrace(
+                requestTraceOptions,
+                this.#client,
+                listOptions
+            );
+            for await (const setting of settings) {
+                if (isFeatureFlag(setting)) {
+                    const flag = parseFeatureFlag(setting);

```

Comment:
What do you think about supporting V2 schema for feature flags as defined here: https://github.com/Azure/AppConfiguration/blob/main/docs/FeatureManagement/FeatureFlag.v2.0.0.schema.json ?

`parseFeatureFlag` returns [FeatureFlagValue](https://learn.microsoft.com/en-us/javascript/api/@azure/app-configuration/featureflagvalue?view=azure-node-latest#@azure-app-configuration-featureflagvalue-conditions) object, which does not have all the fields that we need.

Reply:
Supporting FeatureFlag.v2 schema is on the plan, but is not included in this PR. 
BTW I think it's better to update SDK impl first, as in the latest SDK 1.6.0 the [impl ](https://unpkg.com/@azure/app-configuration@1.6.0/dist/index.js)https://[unpkg.com/@azure/app-configuration@1.6.0/dist/index.js](https://unpkg.com/@azure/app-configuration@1.6.0/dist/index.js)hasn't cover variants yet.

Reply:
SDK updates are usually slow so we dont rely on SDK for schema definitions. We parse the raw value manually in other providers too (example: [Python provider](https://github.com/Azure/azure-sdk-for-python/blob/66d5de409d0980d3b2c7b82971cc648f9d5cd6a5/sdk/appconfiguration/azure-appconfiguration-provider/azure/appconfiguration/provider/_azureappconfigurationprovider.py#L686)).

If we're adding the support for feature flags now, I'd suggest going straight for V2 schema since that's the one we want users to move to eventually.

Reply:
Makes sense. Since we now unified the schema, the config provider is just responsible for providing feature flags. 

BTW after revisiting SDK's impl, I also find another reason that we should not use `FeatureFlagValue` and `parseFeatureFlag`. 
In type `FeatureFlagValue` the properties are renamed from the original schema, following common JS naming convention. e.g. client_filters => clientFilters...etc.

Reply:
now feature flags are parsed directly, v2 schema supported.  8e2de54